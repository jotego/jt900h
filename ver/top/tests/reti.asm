    ; POPSR
    main section code
    org 0
    ld a,0xbf    ; common header

    ld xsp, stack
    ld xwa, end_loop
    push xwa
    ld wa,0x3ff
    push wa
    reti
    jp bad_end

end_loop:
    ld hl,0xbabe
    ld (0xffff),0xff
    jp end_loop
bad_end:
    ld hl,0xdead
    jp bad_end
data:
    dw 0xcafe,0xbeef,0xffff,0xeeee,0xcccc
    dw 0,0,0,0,0,0,0,0,0,0
    align 4
stack:
    end