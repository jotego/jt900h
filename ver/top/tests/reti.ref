000003ff - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 0000babe
-- index registers --
00000000 - 00000000 - 00000000 - 00000058
SR = 8bd7
