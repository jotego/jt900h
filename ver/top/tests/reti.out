000003ff - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 00000000
00000000 - 00000000 - 00000000 - 0000babe
-- index registers --
00000000 - 00000000 - 00000000 - 00000048
SR = 8bd7
